tests/client.test.js

```diff
@@ -49,5 +74,54 @@ const Pulsar = require('../index.js');
         })).toThrow('Service URL is required and must be specified as a string');
       });
     });
+    describe('test getPartitionsForTopic', () => {
+      test('GetPartitions for empty topic', async () => {
+        const client = new Pulsar.Client({
+          serviceUrl: 'pulsar://localhost:6650',
+          operationTimeoutSeconds: 30,
+        });
+
+        await expect(client.getPartitionsForTopic(''))
+          .rejects.toThrow('Failed to GetPartitionsForTopic: InvalidTopicName');
+        await client.close();
+      });
+
+      test('Client/getPartitionsForTopic', async () => {
+        const client = new Pulsar.Client({
+          serviceUrl: 'pulsar://localhost:6650',
+          operationTimeoutSeconds: 30,
+        });
+
+        // test on nonPartitionedTopic
+        const nonPartitionedTopicName = 'test-non-partitioned-topic';
+        const nonPartitionedTopic = `persistent://public/default/${nonPartitionedTopicName}`;
+        await requestAdminApi(`${baseUrl}/admin/v2/persistent/public/default/${nonPartitionedTopicName}`, {
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        });
+        const nonPartitionedTopicList = await client.getPartitionsForTopic(nonPartitionedTopic);
```

Comment:
Before this, you should assert the HTTP response code first.

Reply:
Got it, let me try again.